agent/xds/endpoints.go

```diff
@@ -304,9 +304,8 @@ func (s *ResourceGenerator) endpointsFromSnapshotMeshGateway(cfgSnap *proxycfg.C
 
 		servers, _ := cfgSnap.MeshGateway.WatchedLocalServers.Get(structs.ConsulServiceName)
 		for _, srv := range servers {
-			if isReplica := srv.Service.Meta["read_replica"]; isReplica == "true" {
-				// Peering control-plane traffic can only ever be handled by the local leader.
-				// We avoid routing to read replicas since they will never be Raft voters.
+			isLeader := srv.Service.Meta["leader"] == "true"
```

Comment:
I am not sure if this exists in the metadata

Reply:
Unfortunately, it does not, and adding it would be a chore. The current metadata is constant at startup time so it is propagated with serf, but the raft leader is not known until the leader election completes.
```json
"ServiceMeta": {
  "grpc_port": "8522",
  "grpc_tls_port": "8523",
  "non_voter": "true",
  "raft_version": "3",
  "read_replica": "true",
  "serf_protocol_current": "2",
  "serf_protocol_max": "5",
  "serf_protocol_min": "1",
  "version": "1.16.0"
},
```